perfact/zodbsync/extedit.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+
+from . import helpers
+from .zodbsync import mod_read, mod_write
+
+
+def launch(context, script, path, source=None, orig_source=None):
+    '''
+    Launcher for external edit.
+
+    If called without a source, it is used to create a control file that
+    contains the authentication header found in the current request and the
+    content of the file that can be found under path.
+
+    If a source is provided, updates the object given by path, but only if the
+    current source matches that given in orig_source.
```

Comment:
I would prefer "[...], but only if **its** current source [...]" because there are two kinds of current source (the object's source and current source in the editor).